### ISSUE TITLE
Returned fit causes overflow error

### ISSUE THREAD

**Issue by mayork** [hoburg/gpfit#48]
Again, when trying to fit the compressor maps, I've gotten a number of returned fits with very large exponents. Sometimes, when these are plotted outside of log space, overflow errors occur. I'm not sure this can be avoided, and I don't think the data I'm using is too well conditioned, but it would be nice if there was someway to control how large the exponents were in the fit.

'''
w**0.231 = 0.00187 * (u_1)**-302 * (u_2)**58.2
    + 3.75e-12 * (u_1)**-2.67e+03 * (u_2)**496
    + 0.326 * (u_1)**-7.81 * (u_2)**0.962
    + 0.465 * (u_1)**1.97 * (u_2)**-0.525
'''

**Comment by whoburg** [comment]
Yes! It's called regularization, and we should implement it as an option.

 L2 regularization is trivial to implement. L1 requires more work, but is likely worth it for the better (sparsity inducing) performance. We should implement both.
